src/frontend/src/optimizer/plan_visitor/cardinality_visitor.rs

```diff
@@ -36,11 +36,14 @@ impl CardinalityVisitor {
         input_card: Cardinality,
         eq_set: HashSet<usize>,
     ) -> Cardinality {
-        let mut unique_keys: Vec<HashSet<_>> = vec![input.stream_key().iter().copied().collect()];
-
+        let mut unique_keys: Vec<HashSet<_>> = if let Some(stream_key) = input.stream_key() {
+            vec![stream_key.iter().copied().collect()]
+        } else {
+            vec![]
+        };
```

Comment:
After this PR, it seems we could remove `.filter(|unique_key| !unique_key.is_empty())` this line introduced in https://github.com/risingwavelabs/risingwave/pull/11718 ?

Reply:
done